src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostRawUserInterface.cs

```diff
@@ -1745,10 +1746,10 @@ public override string ToString()
         /// from the keyboard device, blocking processing until a keystroke is
         /// typed that matches the specified keystroke options.
         /// </summary>
-        /// <param name="options">Unused</param>
+        /// <param name="options">Only NoEcho is supported.</param>
         public override KeyInfo ReadKey(ReadKeyOptions options)
         {
-            ConsoleKeyInfo key = Console.ReadKey();
+            ConsoleKeyInfo key = Console.ReadKey(options.HasFlag(ReadKeyOptions.NoEcho));;
```

Comment:
it would be better to get uniform input - one person's opinion vs another. My original code was requested to change to this